### PR TITLE
Add more batch Storage Write API tests

### PR DESCRIPTION
**Pull request by ahmedabu98**

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Run Kotlin_Examples PreCommit

**Comment by ahmedabu98** [comment]
R: @johnjcasey

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java [hunk 1]
```diff
@@ -302,15 +303,15 @@ public void testWriteDynamicDestinations() throws Exception {
   }
 
   @Test
-  public void testWriteDynamicDestinationsBatchWithSchemas() throws Exception {
-    writeDynamicDestinations(true, false);
+  public void testWriteDynamicDestinationsWithAutoSharding() throws Exception {
+    assumeTrue(useStreaming);
+    assumeTrue(!useStorageApiApproximate); // STORAGE_API_AT_LEAST_ONCE ignores auto-sharding
+    writeDynamicDestinations(false, false);
```
Comment (johnjcasey) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:309]: Why did we swap this from true, true to false, false

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java [hunk 2]
```diff
@@ -302,15 +303,15 @@ public void testWriteDynamicDestinations() throws Exception {
...
   }
 
   @Test
-  public void testWriteDynamicDestinationsStreamingWithAutoSharding() throws Exception {
-    assumeTrue(useStreaming);
-    assumeTrue(!useStorageApiApproximate); // STORAGE_API_AT_LEAST_ONCE ignores auto-sharding
-    writeDynamicDestinations(true, true);
+  public void testWriteDynamicDestinationsWithBeamSchemas() throws Exception {
```
Comment (johnjcasey) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:313]: Why did we change the name of this?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java [hunk 3]
```diff
@@ -1329,8 +1330,12 @@ static class SchemaPojo {
   }
 
   @Test
-  public void testSchemaWriteLoads() throws Exception {
-    assumeTrue(!useStreaming);
+  public void testBatchSchemaWriteLoads() throws Exception {
+    // This test is actually for batch!
+    // We just invert useStreaming because we need it as true to test
+    // STORAGE_API_AT_LEAST_ONCE,
+    assumeTrue(useStreaming);
```
Comment (johnjcasey) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:1337]: It looks like you don't invert useStreaming, is that intentional?

Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:1337]: I meant that we're meant to use `!useStreaming` for batch tests, but used `useStreaming` instead. Will update comments to clarify

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java (hunk 2 above)
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:313]: To be more accurate:
1. This tests both batch and streaming modes
2. `withSchemas` actually means using with Beam Schema

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java (hunk 1 above)
Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java:309]: Good catch, this was unintended. Will revert

**Pull request merged**